idseq_dag/steps/run_alignment_remotely.py

```diff
@@ -279,83 +314,83 @@ def run_chunk(self, part_suffix, remote_home_dir, remote_index_dir, remote_work_
+        else:
+            chunk_timeout = int(self.additional_attributes.get(f"{service.lower()}_chunk_timeout", DEFAULT_CHUNK_TIMEOUT))
+            for try_number in range(1, CHUNK_MAX_TRIES + 1):
                 log.write("waiting for {} server for chunk {}. Try #{}".format(
                     service, chunk_id, try_number))
-                max_concurrent = self.additional_attributes["max_concurrent"]
-
-                try:
-                    with server.ASGInstance(service, key_path,
-                                            remote_username, environment,
-                                            max_concurrent, chunk_id,
-                                            self.additional_attributes.get("max_interval_between_describe_instances") or 900,
-                                            self.additional_attributes.get("job_tag_prefix") or "RunningIDseqBatchJob_",
-                                            self.additional_attributes.get("job_tag_refresh_seconds") or 600,
-                                            self.additional_attributes.get("draining_tag") or "draining") as instance_ip:
-                        command.execute(command.remote(commands, key_path, remote_username, instance_ip))
-
-                        if service == "gsnap":
-                            verification_command = "cat %s" % multihit_remote_outfile
+                with server.ASGInstance(service, key_path,
+                                        remote_username, environment,
+                                        self.additional_attributes["max_concurrent"], chunk_id,
+                                        self.additional_attributes.get("max_interval_between_describe_instances, 900),
+                                        self.additional_attributes.get("job_tag_prefix", "RunningIDseqBatchJob_"),
+                                        self.additional_attributes.get("job_tag_refresh_seconds", 600),
+                                        self.additional_attributes.get("draining_tag", "draining") as instance_ip:
+                    # Try/Except block needs to be inside the ASGInstance context.
+                    # A failure to acquire an ASGInstnace is and should be unrecoverable.
+                    elapsed = 0.0
+                    output_corrupt_reason = None
+                    try:
+                        t_start = time.time()
+                        try:
+                            command.execute(command.remote(commands, key_path, remote_username, instance_ip),
+                                            timeout=chunk_timeout)
+                        except:
+                            chunk_status = ChunkStatus.CRASH
+                            raise
+                        finally:
+                            elapsed = time.time() - t_start
+                            if chunk_status == ChunkStatus.CRASH and elapsed >= chunk_timeout:
+                                chunk_status = ChunkStatus.TIMEOUT
```

Comment:
Would it make sense to raise an exception here as well?

Reply:
Ok, I think I got the idea. You are replacing the `CRASH` status by a `TIMEOUT` one.

Reply:
We could refactor `command.execute` to raise a specific exception for timeout conditions. That would error handling more obvious, but this refactoring is clearly out of the scope of this PR.

Reply:
Thanks for the observation, David.   I agree.

Note I just updated the PR to give every chunk it's first try before any chunk gets a retry.  (Well, mostly.  Not a string guarantee.)